data_pull.py

```diff
@@ -53,33 +81,158 @@ def pull_data(self, search_criteria: str, delete_csv: bool = False):
+        :param driver: Webdriver used to connect to Redfin website
+        :param download_dir: Directory where downloaded CSV files will go
+        :param search_criteria: Search criteria to be entered into Redfin search bar (city, state, address, etc.)
+        :param home_types: Type(s) of home of which to pull data for. Options include: house, townhouse, condo, land,
+        multi-family, mobile, co-op, and other. If selecting more than one home type, initialize param as a list
+        :param delete_csv: When set to True, deletes the CSV file that is downloaded as part of the data pulling process
+        :return: Pandas DataFrame containing real estate data from Redfin for the desired city & state
+        """
         # Identify search bar on home page for entering a real estate search location and search for input city/state
         # Finds first search bar on the home page, which is the one we're looking for
-        driver.find_element(By.CLASS_NAME, "search-input-box").send_keys(f"{search_criteria}" + Keys.ENTER)
-        driver.implicitly_wait(1)
+        driver.find_element(By.CLASS_NAME, 'search-input-box').send_keys(f'{search_criteria}' + Keys.ENTER)
+        time.sleep(1.5)
+
+        # Click 'All Filters' and select 'Sold' filter
+        driver.find_element(By.XPATH, '//*[@id="sidepane-header"]/div/div[1]/form/div[5]/div').click()
+        driver.find_element(By.XPATH, '//*[@id="filterContent"]/div/div[1]/div/div/div/div/div/div/div[3]').click()
+        time.sleep(1.5)
+
+        # Select desired 'Home Type'
+        home_type_header = driver.find_element(By.XPATH, '//*[@id="filterContent"]/div/div[5]/div[1]/div/span')
+        driver.execute_script("return arguments[0].scrollIntoView();", home_type_header)
+        time.sleep(1.5)
+
+        if type(home_types) == str:     # If user selects single home type (string), convert it to a list
+            home_types = [home_types]
+        home_type_select(home_types=home_types, driver=driver)
+        time.sleep(1.5)
+
+        # Set to only include sales from the last week (reduces # of homes to download; need to keep < 350)
+        select = Select(driver.find_element(By.XPATH, '//*[@id="filterContent"]/div/div[6]/div[2]/span/span/select'))
+        select.select_by_visible_text('Last 1 week')
+        time.sleep(1.5)
+
+        # Turn off 'Foreclosures' listing type
+        listing_type_header = driver.find_element(By.XPATH, '//*[@id="filterContent"]/div/div[10]/div[1]/div/span')
+        driver.execute_script('return arguments[0].scrollIntoView()', listing_type_header)
+        time.sleep(1.5)
+
+        driver.find_element(By.XPATH, '//*[@id="filterContent"]/div/div[10]/div[2]/div/div[1]/div[2]/span/label/span[1]').click()
+        time.sleep(1.5)
+
+        # Close 'All Filters' menu
+        driver.find_element(By.XPATH, '//*[@id="right-container"]/div[6]/div/aside/header/button').click()
+        time.sleep(1.5)
```

Comment:
I think I have a good enough handle on the debugger to get good use out of it, but that's not why I have so many of the waits. I put them there as a sort of failsafe to give the page more time to load the changes I was making at each step. It would run without the waits (most times), but most of the time at the end the page wouldn't be done applying the change I made by clicking a filter before downloading the data. I'd end up getting a bunch of extra records I didn't want, and with a cap of 350 records per download it would hit that cap pretty frequently otherwise.